File: src/main/java/com/example/jobs/entity/Developer.java
package com.example.jobs.entity;

import java.time.OffsetDateTime;
import java.util.Collections;
import java.util.List;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "developers")
public class Developer implements Persistable<UUID> {
  @Id
  @Column("id")
  private UUID devId;

  @Transient
  private boolean isNewEntity = true;

  @Column("name")
  private String name = "";

  @Column("email")
  private String email = "";

  @Column("api_key")
  private String apiKey = "";

  @Column("rate_limit_per_hour")
  private int rateLimitPerHour = 100;

  @Column("is_active")
  private boolean isActive = true;

  @Column("tier")
  private String tier = "free";

  @Column("created_at")
  private OffsetDateTime createdAt = OffsetDateTime.now();

  @Transient
  private List<String> permissions = Collections.emptyList();

  @Override
  public UUID getId() { return this.devId; }

  @Override
  public boolean isNew() { return this.isNewEntity; }

  public void setId(UUID id) { this.devId = id; }
  public void setNew(boolean isNew) { this.isNewEntity = isNew; }
}

File: src/main/kotlin/com/example/jobs/entity/Developer.kt
package com.example.jobs.entity

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.OffsetDateTime
import java.util.UUID

@Table(name = "developers")
class Developer : Persistable<UUID> {

    @Id
    @Column("id")
    private var devId: UUID? = null

    @Transient
    private var isNewEntity: Boolean = true

    @Column("name")
    var name: String = ""

    @Column("email")
    var email: String = ""

    @Column("api_key")
    var apiKey: String = ""

    @Column("rate_limit_per_hour")
    var rateLimitPerHour: Int = 100

    @Column("is_active")
    var isActive: Boolean = true

    @Column("tier")
    var tier: String = "free"

    @Column("created_at")
    var createdAt: OffsetDateTime = OffsetDateTime.now()

    @Transient
    var permissions: List<String> = emptyList()

    override fun getId(): UUID? = this.devId
    override fun isNew(): Boolean = this.isNewEntity

    fun setId(id: UUID) {
        this.devId = id
    }

    fun setNew(isNew: Boolean) {
        this.isNewEntity = isNew
    }
}